### PR TITLE
Upgrade Gardener to 1.10.1

### PR DESCRIPTION
**Pull request by rafalpotempa**
**Description**

Changes proposed in this pull request:

- Upgrade Gardener to 1.10.1 (version from `master`)

**Related issue(s)**

### CODE REVIEW

**Comment by franpog859** [comment]
The Upgrade Shoot test failed with the timeout:
```
time="2020-08-31T12:24:05Z" level=info msg="Asserting operation 527b6010-ab41-44cf-a84b-424b03c5965e is in Succeeded state."
time="2020-08-31T12:24:05Z" level=info msg="Operation message: error: timeout while processing operation"
    operation.go:26:
                Error Trace:    operation.go:26
                                                        operation.go:13
                                                        upgrade_shoot_test.go:76
                Error:          Not equal:
                                expected: "Succeeded"
                                actual  : "Failed"
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,2 @@
                                -(gqlschema.OperationState) (len=9) Succeeded
                                +(gqlschema.OperationState) (len=6) Failed
                Test:           TestShootUpgrade/Azure
```
I'm retriggering the job
/test pre-master-control-plane-gke-provisioner-integration

**Comment by rafalpotempa** [comment]
There is still problem with this KEB thing

**Comment by rafalpotempa** [comment]
And it's gone :)

**Comment by franpog859** [comment]
The provisioning test failed:
```
=== CONT  Test_E2E_Gardener/Azure
    helpers.go:12:
                Error Trace:    helpers.go:12
                                                        e2e_test.go:51
                Error:          Received unexpected error:
                                Failed to receive proper Operation Status response
                                github.com/kyma-project/control-plane/tests/provisioner-tests/test/testkit/compass/provisioner.client.ProvisionRuntime
                                        /go/src/github.com/kyma-project/control-plane/tests/provisioner-tests/test/testkit/compass/provisioner/client.go:56
                                github.com/kyma-project/control-plane/tests/provisioner-tests/test/provisioner.Test_E2E_Gardener.func1.1
                                        /go/src/github.com/kyma-project/control-plane/tests/provisioner-tests/test/provisioner/e2e_test.go:50
                                testing.tRunner
                                        /usr/local/go/src/testing/testing.go:1039
                                runtime.goexit
                                        /usr/local/go/src/runtime/asm_amd64.s:1373
                Test:           Test_E2E_Gardener/Azure
    helpers.go:16: Received unexpected error: Failed to receive proper Operation Status response [Error while starting Runtime provisioning]
```
this error seems to be flaky. I'll retest the job
/test pre-master-control-plane-gke-provisioner-integration

**Comment by franpog859** [comment]
It happened again. @rafalpotempa , could you take a look at it? 😢

**Comment by rafalpotempa** [comment]
/test 
pre-master-control-plane-gke-provisioner-integration

File: components/provisioner/go.mod [hunk 1]
```diff
@@ -5,45 +5,42 @@ go 1.14
...
-	k8s.io/apiextensions-apiserver v0.17.2 => k8s.io/apiextensions-apiserver v0.15.8-beta.1
-	k8s.io/apimachinery v0.17.2 => k8s.io/apimachinery v0.15.8-beta.1
-	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible => k8s.io/client-go v0.15.8-beta.1
-	sigs.k8s.io/controller-runtime v0.5.0 => sigs.k8s.io/controller-runtime v0.3.0
+	golang.org/x/crypto => golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a
+
+	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible => k8s.io/client-go v0.18.8
 )
+
+replace github.com/kyma-project/control-plane => github.com/rafalpotempa/control-plane v0.0.0-20200903082609-3c784a9a689c
```
Comment (akgalwas) [inline components/provisioner/go.mod:46]: Consider updating control-plane dependency to some more recent commit.

**Review by akgalwas — APPROVED** [review]

**Review by akgalwas — APPROVED** [review]

**Pull request merged**